requirements.yml

```diff
@@ -18,6 +18,7 @@ collections:
   - ansible.posix
   - community.general
   - containers.podman
+  - openstack.cloud
```

Comment:
what's the reason for this? we don't want to use the openstack collection anywhere in edpm-ansible. we need to move away from that.

Reply:
It's [still used](https://github.com/openstack-k8s-operators/edpm-ansible/blob/67995fb067b069751d55f358c5541e9d96222bbb/edpm_ansible/ansible_plugins/modules/edpm_ovn_mac_addresses.py#L27-L29) in `edpm_ovn_mac_addresses` module, which is used by `edpm_ovn` role. We could think about importing only the required functions if they are needed.

Reply:
The functions are:

- [openstack_full_ansible_spec](https://github.com/openstack/ansible-collections-openstack/blob/edd4e1b2e944370bc22dcfd75f8e41d886811fec/plugins/module_utils/openstack.py#L120-L146)
- [openstack_module_kwargs](https://github.com/openstack/ansible-collections-openstack/blob/edd4e1b2e944370bc22dcfd75f8e41d886811fec/plugins/module_utils/openstack.py#L149-L157)
- [openstack_cloud_from_modules](https://github.com/openstack/ansible-collections-openstack/blob/edd4e1b2e944370bc22dcfd75f8e41d886811fec/plugins/module_utils/openstack.py#L161-L207)

if we choose to import them we also add `openstacksdk` as dependency.

Reply:
It's used in edpm_ansible/roles/edpm_ovn/tasks/externaldeploy.yml which came from external_deploy_tasks in deployment/ovn/ovn-controller-container-puppet.yaml from tripleo-heat-templates. That's not something we're going to use in edpm-ansible as it actually talks to neutron to allocate ports in the undercloud neutron. So, that doesn't fit the edpm architecture.

We may need something to still generate MAC's and populate the vars file that the module was doing (ovn_bridge_mac_address_vars.yaml). But, that won't need to talk to Neutron, so we can do without importing the openstack collection or openstacksdk in this module. @hjensas any thoughts about that?

Reply:
Totally agree we should not use neutron to allocate ports for OVN bridges. But AFIK we need something to generate MAC addresses and keep track that the MAC for each host:bridge is uniq. May need to create a helper service for this, some simple python rest service keep track of used MAC's and which host[bridge] mac belongs to? Or we might be able to create a kubernetes operator/controller? that would use the kubernetes etcd as the persistent store?

Reply:
in ospdo we created this controller [1] to create unique mac addresses. We might we want to add something similar to the infra-operator. 

[1] https://github.com/openstack-k8s-operators/osp-director-operator/blob/master/controllers/openstackmacaddress_controller.go